src/mir/index.md

```diff
@@ -97,11 +93,27 @@ let mut _5: &mut std::vec::Vec<i32>;
 
 You can see that variables in MIR don't have names, they have indices,
 like `_0` or `_1`.  We also intermingle the user's variables (e.g.,
-`_1`) with temporary values (e.g., `_2` or `_3`). You can tell the
-difference between user-defined variables have a comment that gives
-you their original name (`// "vec" in scope 1...`). The "scope" blocks
-(e.g., `scope 1 { .. }`) describe the lexical structure of the source
-program (which names were in scope when).
+`_1`) with temporary values (e.g., `_2` or `_3`). You can tell apart
+user-defined variables because they have debuginfo associated to them (see below).
+
+**User variable debuginfo.** Below the variable declarations, we find the only
+hint that `_1` represents a user variable:
+```mir
+scope 1 {
+    debug vec => _1;                 // in scope 1 at src/main.rs:2:9: 2:16
+}
+```
+Each `debug <Name> => <Place>;` annotation describes a named user variable,
+and where (i.e. the place) a debugger can find the data of that variable.
+Here the mapping is trivial, but optimizations may complicate the place,
+or lead to multiple user variables sharing the same place.
+Additionally, closure captures are described using the same system, and so
+they're complicated even without optimizations, e.g.: `debug x => (*((*_1).0: &T));`.
```

Comment:
it would be good, I think, to have some links to the data structures that describe this in the MIR. I dont' thnk we have to go into details, but just a few links would help people know where to get started.

Reply:
I agree, but it feels like this introduction to MIR is trying to stay light, and that makes it awkward to go into details.

Reply:
yeah, really I think what we need is to add a subsection with more details. For now, I'm going to merge this PR as is.

Reply:
MIR feels like it should be a chapter with sections for every part, perhaps including thing like semantic constraints. I don't know why I expected that to already be a thing at first, I guess I haven't kept up with `rustc-guide`.